src/memory.rs

```diff
@@ -32,7 +33,11 @@ impl MemoryManager {
             if let Some((idx, _)) = allocs.iter().find_position(|i| !(**i)) {
                 // we found an unallocated buffer, claim it and make a handle to it
                 allocs[idx] = true;
-                break MemoryHandle { len: 0, allocation_idx: idx, mem: BUFFERS[idx].as_ptr() as *mut [u8; BUFFER_SIZE] };
```

Comment:
Oh, so this was what caused that segfault?

Reply:
The definition of buffers not being mut